### ISSUE TITLE
Edge 130 does only report battery in steps of 20 which yields zero burn rate

### ISSUE THREAD

**Issue by matherm** [bfkeinberg/batteryBurnRate#1]
First of all thanks for the nice idea. I was just about starting to create my own data field and then came across yours.

I have a Edge 130 with latest 3.70 installed.

Unfortunately the Edge 130 only reports battery levels in steps of 20 (100,80,60,40,20).
I thought it may be possible to extend the calculations to this case, too.

My idea was to fit a straight line through some last values and compute the slope, which should give an estimator for burining-rate and battery dead.

I have some sample code, but I'm not sure whether to fork or to integrate.

Are you interested?

Best wishes
Matthias

**Comment by bkfeinberg** [comment]
Oh, you could definitely send me a PR. I don't have any way to test, so you'd need to sideload on your device to see if it works. I assume there's a data structure to look at to see your exact device model to enable the code.

**Comment by matherm** [comment]
Great, sounds very good!

I integrated the code into your master branch and i's now running on my device.
For simplicity I just created a user-editable setting "regressionMode" and implemented a switch.

I'll test it on my device over night and will send you a PR if I'm satisfied.

**Comment by matherm** [comment]
I'm pretty happy with the test and the calculated values are much more helpful than the one of the standard battery badge. 

In short: 
* I implemented a fixed size ring buffer over an estimated overall battery life (e.g. 10 h, is user-configureable).
* The buffer fills over time with values
* When at least two values are available the regression starts
* Output is an estimate of the current burn rate/h and an estimated ETA for remaining battery life

This should work for other devices with fine-grained battery values, too. But I just have this tiny edge 130 for testing.

I have sent you a PR.

**Comment by bkfeinberg** [comment]
I will try the same as you on my device; I will install the build and let it run overnight. If ok then we can merge.

**Comment by bkfeinberg** [comment]
I just tested overnight on my device with the new mode off, now I will
leave it running with the new mode on.

**Comment by bkfeinberg** [comment]
Hmm, on my 830, the text in the new mode isn't readable. Screenshot attached. There must be a way to scale the text?


![EdgeScreenshot](https://user-images.githubusercontent.com/9753972/85598939-e5ce9e00-b600-11ea-82df-cc3b57efe669.jpg)

**Comment by bkfeinberg** [comment]
I wonder if this requires changing this field from a SimpleDataField to a DataField. That would allow for setting the font size.

**Comment by matherm** [comment]
Ouff, I'm sorry. I don't want you to have too much trouble with it.

I'm not sure how many edge 130 users there are at all and I just saw that the new edge 130 plus has battery value percentages, again.

I'm not sure about the UI-scaling. I have to look it up.

What do you think?

**Comment by bkfeinberg** [comment]
That's interesting, so the api gives you reasonable values now? Tell you
what, why don't you try my latest version and see how it works for you. I
assume that the previous versions would
not have even started on a 130 due to memory issues. I don't think it's bad
that I went ahead and changed it into a DataField proper, that gives
flexibility in layout for the future.

If my current version still isn't good enough on your device, maybe try out
your branch again, now that I fixed the merge conflicts in it?
